packages/react-server/src/ReactFizzServer.js

```diff
@@ -97,119 +212,580 @@ function renderNode(request: Request, node: ReactNodeList): void {
+    // on preparing fallbacks if we don't have any more main content to work on.
+    request.pingedWork.push(suspendedFallbackWork);
+
+    // This segment is the actual child content. We can start rendering that immediately.
+    const contentRootSegment = createPendingSegment(request, 0, null);
+    // We mark the root segment as having its parent flushed. It's not really flushed but there is
+    // no parent segment so there's nothing to wait on.
+    contentRootSegment.parentFlushed = true;
+    // TODO: Currently this is running synchronously. We could instead schedule this to pingedWork.
+    // I suspect that there might be some efficiency benefits from not creating the suspended work
+    // and instead just using the stack if possible. Particularly when we add contexts.
+    const contentWork = createSuspendedWork(
+      request,
+      content,
+      newBoundary,
+      contentRootSegment,
+      null,
+    );
+    retryWork(request, contentWork);
   } else {
     throw new Error('Not yet implemented element type.');
   }
 }
 
-function retrySegment(request: Request, segment: Segment): void {
+function errorWork(
+  request: Request,
+  boundary: Root | SuspenseBoundary,
+  segment: Segment,
+  error: mixed,
+) {
+  segment.status = ERRORED;
+
+  request.allPendingWork--;
+  if (boundary !== null) {
+    boundary.pendingWork--;
+  }
+
+  // Report the error to a global handler.
+  reportError(request, error);
+  if (boundary === null) {
+    fatalError(request, error);
+  } else if (!boundary.forceClientRender) {
+    boundary.forceClientRender = true;
+    // Regardless of what happens next, this boundary won't be displayed,
+    // so we can flush it, if the parent already flushed.
+    if (boundary.parentFlushed) {
+      // We don't have a preference where in the queue this goes since it's likely
+      // to error on the client anyway. However, intentionally client-rendered
+      // boundaries should be flushed earlier so that they can start on the client.
+      // We reuse the same queue for errors.
+      request.clientRenderedBoundaries.push(boundary);
+    }
+  }
+}
+
+function completeWork(
+  request: Request,
+  boundary: Root | SuspenseBoundary,
+  segment: Segment,
+) {
+  segment.status = COMPLETED;
+  request.allPendingWork--;
+
+  if (boundary === null) {
+    request.pendingRootWork--;
+    if (segment.parentFlushed) {
+      invariant(
+        request.completedRootSegment === null,
+        'There can only be one root segment. This is a bug in React.',
+      );
+      request.completedRootSegment = segment;
+    }
+    return;
+  }
+
+  boundary.pendingWork--;
+  if (boundary.forceClientRender) {
+    // This already errored.
+    return;
+  }
+  if (boundary.pendingWork === 0) {
+    // This must have been the last segment we were waiting on. This boundary is now complete.
+    if (segment.parentFlushed) {
+      // Our parent segment already flushed, so we need to schedule this segment to be emitted.
```

Comment:
Is "flushed" and "emitted" the same thing?

Reply:
Yea. I guess emitted is a bit more vague because it's a process of also flushing its children. Not just flushing this segment.